Classes/Console/Command/Cache/CacheFlushCommand.php

```diff
@@ -0,0 +1,126 @@
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+class CacheFlushCommand extends AbstractConvertedCommand
+{
+    protected function configure()
+    {
+        $this->setDescription('Flush all caches');
+        $this->setHelp(
+            <<<'EOH'
+Flushes TYPO3 core caches first and after that, flushes caches from extensions.
+EOH
+        );
+
+        $this->setDefinition($this->createCompleteInputDefinition());
+    }
+
+    protected function createNativeDefinition(): array
+    {
+        return [
+            new InputOption(
+                'files-only',
+                null,
+                InputOption::VALUE_NONE,
+                'Only file caches are flushed'
+            ),
+        ];
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $filesOnly = $input->getOption('files-only');
+        $application = $this->getApplication();
+        if (!$application instanceof Application) {
+            throw new \RuntimeException('Fatal error. Application is not properly initialized.', 1546617606);
+        }
+        $filesOnly = $filesOnly || !$application->isFullyCapable();
+
+        $io = new SymfonyStyle($input, $output);
+
+        $lowLevelCleaner = new CacheLowLevelCleaner();
+        $lowLevelCleaner->forceFlushCachesFiles();
+        if ($filesOnly) {
+            $io->writeln('Flushed all file caches.');
+            // No need to proceed, as files only flush is requested
+            return;
+        }
+
+        $lowLevelCleaner->forceFlushDatabaseCacheTables();
+        $application->boot(RunLevel::LEVEL_FULL);
+
+        $cacheService = new CacheService();
+        $cacheService->flush();
+        $cacheService->flushCachesWithDataHandler();
+
+        $io->writeln('Flushed all caches.');
+    }
+
+    /**
+     * @deprecated will be removed with 6.0
+     *
+     * @return array
+     */
+    protected function createDeprecatedDefinition(): array
+    {
+        return [
+            new InputOption(
+                'force',
+                null,
+                InputOption::VALUE_NONE,
+                'Cache is forcibly flushed (low level operations are performed)'
+            ),
+            new InputArgument(
+                'force',
+                null,
+                'Cache is forcibly flushed (low level operations are performed)',
+                false
+            ),
+            new InputArgument(
+                'filesOnly',
+                null,
+                'Only file caches are flushed',
+                false
+            ),
+        ];
+    }
+
+    /**
+     * @deprecated will be removed with 6.0
+     */
+    protected function handleDeprecatedArgumentsAndOptions(InputInterface $input, OutputInterface $output)
+    {
+        if ($input->getArgument('force')
+        || $input->getArgument('filesOnly')
+        || $input->getOption('files-only')
+        || $input->getOption('force')
```

Comment:
Bye bye `--force`, you will be missed. ;-)

Reply:
@mbrodala Do you see any issues with that? Thought it would make it simpler for all of us, as I realized, most people use this option anyway and even TYPO3 install tool cache flush does it this way.

Reply:
No issue. It's just that it has entered muscle memory for many people. ;-)